src/server/game/Scripting/ScriptMgr.cpp

```diff
@@ -96,6 +96,10 @@ template<>
 struct is_script_database_bound<AchievementCriteriaScript>
     : std::true_type { };
 
+template<>
+struct is_script_database_bound<AreaTriggerEntityScript>
```

Comment:
`AreaTriggerEntityScript` also need a specialization of `ScriptRegistrySwapHooks` to be swapped correctly when using script hotswapping (othwerise it will cause crashes since unloaded scripts are still referenced).

If you can't make it specialize it as:

```c++
template<typename Base>
class ScriptRegistrySwapHooks<AreaTriggerEntityScript, Base>
    : public UnsupportedScriptRegistrySwapHooks<Base> { };
```

and I will take care of it before it gets merged.